src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,103 @@
-export const Pagination = () => {};
+import React, { useEffect, useState } from 'react';
+import cn from 'classnames';
+import { getNumbers } from '../../utils';
+
+type Props = {
+  total: number;
+  perPage: number;
+  currentPage: number;
+  onPageChange: (page: number) => void;
+};
+
+export const Pagination: React.FC<Props> = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  const [chosenPage, setChosenPage] = useState(currentPage);
+
+  useEffect(() => {
+    onPageChange(chosenPage);
+  }, [chosenPage]);
+
+  useEffect(() => {
+    setChosenPage(currentPage);
+  }, [perPage]);
```

Comment:
I suppose you should do these actions on click, the useEffect is not the best approach. Just call this function on click